"make deploy-kind" couldn't quite set up controller-gen

I have a fresh Arch Linux install on which I have set up kind. I cloned this repo, then ran the following commands:

```
$ kubectl cluster-info --context kind-kind
Kubernetes master is running at https://127.0.0.1:45357
KubeDNS is running at https://127.0.0.1:45357/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
lbakken@shostakovich ~/development/pivotal/rabbitmq-for-kubernetes (master=)
$ make deploy-kind
which: no controller-gen in (/home/lbakken/.kiex/elixirs/elixir-1.10.3/bin:/home/lbakken/.kiex/bin:/home/lbakken/.kiex/elixirs/elixir-1.10.3/bin:/home/lbakken/.kiex/bin:/home/lbakken/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/lbakken/go/bin:/home/lbakken/.cache/rebar3/bin:/home/lbakken/go/bin:/home/lbakken/.cache/rebar3/bin)
updating kustomize image patch file for manager resource
sed -i'' -e 's@image: .*@image: '"dev.registry.pivotal.io/p-rabbitmq-for-kubernetes/rabbitmq-for-kubernetes-operator:"d95e96d-""'@' ./config/default/overlays/kind/manager_image_patch.yaml
go: creating new go.mod: module tmp
go: downloading sigs.k8s.io/controller-tools v0.2.5
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.2.5
go: downloading github.com/spf13/cobra v0.0.5
go: downloading sigs.k8s.io/yaml v1.1.0
go: downloading k8s.io/api v0.17.0
go: downloading k8s.io/apiextensions-apiserver v0.17.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20190905181640-827449938966
go: downloading k8s.io/apimachinery v0.17.0
go: downloading github.com/gobuffalo/flect v0.2.0
go: downloading golang.org/x/tools v0.0.0-20190920225731-5eefd052ad72
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading k8s.io/utils v0.0.0-20191114184206-e782cd3c129f
go: downloading github.com/gogo/protobuf v1.2.2-0.20190723190241-65acae22fc9d
go: downloading gopkg.in/yaml.v2 v2.2.4
go: downloading golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/fatih/color v1.7.0
go: downloading k8s.io/klog v1.0.0
go: downloading github.com/mattn/go-isatty v0.0.8
go: downloading github.com/google/gofuzz v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.2
go: downloading golang.org/x/text v0.3.2
go: downloading golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456
/controller-gen "crd:trivialVersions=true, preserveUnknownFields=false" rbac:roleName=operator-role paths="./api/...;./controllers/..." output:crd:artifacts:config=config/crd/bases
/bin/sh: /controller-gen: No such file or directory
make: *** [Makefile:29: manifests] Error 127
```

Re-running `make deploy-kind` works as the expected binary is there:

```
lbakken@shostakovich ~/development/pivotal/rabbitmq-for-kubernetes (master *=)
$ which controller-gen
/home/lbakken/go/bin/controller-gen
```

 - OS: Arch Linux, updated as of 2020-05-19

--- a/Makefile
+++ b/Makefile
@@ -189,6 +189,9 @@ ifeq (, $(shell which controller-gen))
 	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.5 ;\
 	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
 	}
+ifeq (, $(GOBIN))
+GOBIN=$(GOPATH)/bin
+endif
 CONTROLLER_GEN=$(GOBIN)/controller-gen
 else
 CONTROLLER_GEN=$(shell which controller-gen)

